### PR TITLE
Add Mauritania holidays

### PR DESCRIPTION
**Pull request by kasya**
## Proposed change

Add Mauritania holidays. 

## Type of change

- [x] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/dev/docs/source

### CODE REVIEW

File: holidays/countries/mauritania.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        super()._populate(year)
+
+        # New Year's Day.
+        self._add_new_years_day("New Year's Day")
+
+        # Labor Day.
+        self._add_labor_day("Labor Day")
+
+        # Africa Day.
+        self._add_holiday_may_25("Africa Day")
```
Comment (KJhellico) [inline holidays/countries/mauritania.py:55]:
```suggestion
        self._add_africa_day("Africa Day")
```

File: tests/countries/test_mauritania.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        self.assertHolidayName("New Year's Day", (f"{year}-01-01" for year in range(1950, 2050)))
+
+    def test_labor_day(self):
+        self.assertHolidayName("Labor Day", (f"{year}-05-01" for year in range(1950, 2050)))
+
+    def test_africa_day(self):
+        self.assertHolidayName("Africa Day", (f"{year}-05-25" for year in range(1950, 2050)))
+
+    def test_independence_day(self):
+        self.assertHolidayName("Independence Day", (f"{year}-11-28" for year in range(1960, 2050)))
```
Comment (KJhellico) [inline tests/countries/test_mauritania.py:48]:
```suggestion
        name = "Independence Day"
        self.assertHolidayName(name, (f"{year}-11-28" for year in range(1960, 2050)))
        self.assertNoHolidayName(name, range(1950, 1960))
```

**Review by KJhellico — COMMENTED** [review]
@kasya, it's great work! There hasn't been new countries for a long time. Please take a look at few suggestions.

**Review by arkid15r — COMMENTED** [review]
LGTM,
thank you @kasya

**Review by KJhellico — APPROVED** [review]
LGTM.

**Pull request merged**